docker/bin/km-local

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+
+# Run etcd, mesos-local, km apiserver, km controller-manager, and km scheduler.
+# Exiting (or failure of) this script will cause the above sub-processes to be killed.
+# Usage: km-local
+# Mac support requires gtimeout & pcregrep (brew install coreutils && brew install pcre)
+# Inputs:
+# PUBLIC_IP - IP the km components will listen on
+# MESOS_IP - IP that mesos is accessible on
+# MESOS_PORT - Port that mesos-master is accessible on (default 5050)
+# ETCD_IP - IP that etcd is accessible on
+# ETCD_PORT - Port that etcd is accessible on (default 4001)
+# LOG_DIR - Directory to write logs to (default /tmp/k8sm-logs)
+
+set -e
+
+# add the current dir to PATH
+bin=$(cd $(dirname $0) && pwd -P)
+export PATH=$PATH:${bin}
+
+# validate required scripts
+cmd-exists "await-service"
+
+# add kubernetes-mesos/bin to PATH (unless km-local has been moved)
+k8sm_bin=$(cd ${bin}/../../bin && pwd -P)
+[ -d "$k8sm_bin" ] && [ -e "${k8sm_bin}/km" ] && export PATH=$PATH:${k8sm_bin} || true
+
+# validate required binaries
+cmd-exists "km"
+
+TIMEOUT=$(which timeout || true)
+if ! cmd-exists "timeout"; then
+    # mac homebrew installs gnu coreutils with g- prefix
+    ! cmd-exists "gtimeout" && echo "timeout or gtimeout command not found" && exit 1
+    TIMEOUT=$(which gtimeout)
+fi
+
+PUBLIC_IP=$(publicip)
+
+set -e
+
+export K8SM_IP=${PUBLIC_IP}
+export K8SM_PORT=8888
+echo "Kubernetes: ${K8SM_IP}:${K8SM_PORT}"
+
+export MESOS_IP=${MESOS_IP:-$PUBLIC_IP}
```

Comment:
I'm extracting the common functions to a `subprocs.sh` and hiding that timeout mess in `timeout`, but I'm not sure I want to extract the IPs, namely because they choose default values which I kinda hate doing in shell scripts. If there were a way to not have to hardcode defaults at all but instead pull the values from the running commands and use the command's defaults when launching them... that would be far superior.